prog/target.go

```diff
@@ -53,6 +53,8 @@ type Target struct {
 	SyscallMap  map[string]*Syscall
 	ConstMap    map[string]uint64
 	resourceMap map[string]*ResourceDesc
+	// Maps calls to resource desc ptrs for resources they create.
+	resourceDescMap map[*Syscall][]*ResourceDesc
```

Comment:
We use this resourceDescMap only to populate another cache -- resourceCtors below.
So it looks wrong to store resourceDescMap here, it's needed only for the duration of initTarget while it fills resourceCtors.
But if this is slow, I think we should calculate resourceCtors statically during compilation of descriptions in pkg/compiler.
E.g. add something like:
```
@@ -132,6 +132,7 @@ type ResourceDesc struct {
        Type   Type
        Kind   []string
        Values []uint64
+       Ctors  []ResourceCtor
 }

type ResourceCtor struct {
    Call int // Index in Target.Syscalls
    Precise bool
}
```
and fill it in pkg/compiler (we calculate ctors there already anyway).

Reply:
Looks like we only properly populate resources in `initTarget()`, so I moved all precomputation there and changed it to use the approach with `ResourceCtor`.